javaagent-exporters/prometheus/src/main/java/io/opentelemetry/javaagent/exporters/prometheus/PrometheusMetricCollector.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.opentelemetry.javaagent.exporters.prometheus;
+
+import com.google.auto.service.AutoService;
+import io.opentelemetry.exporters.prometheus.PrometheusCollector;
+import io.opentelemetry.javaagent.spi.exporter.ExporterConfig;
+import io.opentelemetry.javaagent.spi.exporter.MetricCollector;
+import io.opentelemetry.sdk.metrics.export.MetricProducer;
+import io.prometheus.client.exporter.HTTPServer;
+import java.io.IOException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+@AutoService(MetricCollector.class)
+public class PrometheusMetricCollector implements MetricCollector {
+  private static final Logger log = LoggerFactory.getLogger(PrometheusMetricCollector.class);
+
+  private static final String EXPORTER_NAME = "prometheus";
+  private static final String PORT_CONF_PROP_NAME = EXPORTER_NAME + ".port";
+  private static final String HOST_CONF_PROP_NAME = EXPORTER_NAME + ".host";
+  private static final int DEFAULT_PORT = 9464;
+  // The empty address equals to any address
+  private static final String DEFAULT_HOST = "0.0.0.0";
+
+  @Override
+  public void init(MetricProducer producer, ExporterConfig config) {
```

Comment:
I think this method requires documentation to explain what is going on here. Just reading it I don't understand what is a connection between `PrometheusCollector` and we built and registered and `HTTPServer` that we started later. How that server gets metrics?

The absence of javadoc on `PrometheusCollector` does not help it neither :)

Reply:
+1 I will add some docs.